src/LightStep/Collector/LightStepHttpClient.cs

```diff
@@ -32,7 +32,37 @@ public LightStepHttpClient(string url, Options options)
         {
             _url = url;
             _options = options;
-            _client = new HttpClient {Timeout = _options.ReportTimeout};
+            _client = new HttpClient() {Timeout = _options.ReportTimeout};
+        }
+
+        internal HttpRequestMessage CreateStringRequest(ReportRequest report)
+        {
+            var jsonFormatter = new JsonFormatter(new JsonFormatter.Settings(true));
+            var jsonReport = jsonFormatter.Format(report);
+            var request = new HttpRequestMessage(HttpMethod.Post, _url)
+            {
+                Version = _options.UseHttp2 ? new Version(2, 0) : new Version(1, 1),
+                Content = new StringContent(jsonReport)
+            };
+            request.Content.Headers.ContentType = new MediaTypeHeaderValue("application/json");
+            return request;
+        }
+
+        internal HttpRequestMessage CreateBinaryRequest(ReportRequest report)
+        {
+            var binaryReport = report.ToByteArray();
+            var request = new HttpRequestMessage(HttpMethod.Post, _url)
+            {
+                Version = _options.UseHttp2 ? new Version(2, 0) : new Version(1, 1),
```

Comment:
Aside: how are downgrades from 2 to 1.1 handled? Could we always use HTTP2 and have the downgrade handled for us?

Reply:
The problem is that certain .NET installations won't have HTTP/2 available, and trying to use it will cause a runtime error. If it _is_ available, the framework will gracefully fallback from HTTP/2 if the server doesn't support it; I'll add a note to the README to indicate that users are encouraged to set `UseHttp2` on modern .NET installs.